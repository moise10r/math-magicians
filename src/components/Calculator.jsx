import React, { Component } from 'react';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
    this.rows = {
      row1: ['AC', '+/-', '%', 'รท'],
      row2: ['7', '8', '9', 'x'],
      row3: ['4', '5', '6', '-'],
      row4: ['1', '2', '3', '+'],
      row5: ['0', '.', '='],
    };
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  render() {
    const { total, next, operation } = this.state;
    console.log(total, next, operation);
    return (
      <div className="calculator-main-container">
        <div className="row flex-center result">
          <input type="text" />
        </div>
        <div className="row flex-center ">
          {this.rows.row1.map((key) => <button onClick={() => this.handleKeyPress(key)} key={key} className="btn" type="button">{key}</button>)}
        </div>
        <div className="row flex-center ">
          {this.rows.row2.map((key) => <button onClick={() => this.handleKeyPress(key)} key={key} className="btn" type="button">{key}</button>)}
        </div>
        <div className="row flex-center ">
          {this.rows.row3.map((key) => <button onClick={() => this.handleKeyPress(key)} key={key} className="btn" type="button">{key}</button>)}
        </div>
        <div className="row flex-center ">
          {this.rows.row4.map((key) => <button onClick={() => this.handleKeyPress(key)} key={key} className="btn" type="button">{key}</button>)}
        </div>
        <div className="row flex-center ">
          {this.rows.row5.map((key) => <button onClick={() => this.handleKeyPress(key)} key={key} className="btn" type="button">{key}</button>)}
        </div>
      </div>
    );
  }
}

export default Calculator;
