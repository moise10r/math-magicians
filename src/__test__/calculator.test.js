import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator page component', () => {
  it('should match the Calculator page', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return the sum of two number form UI', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('='));
    const result = screen.getByRole('textbox');
    expect(result).toHaveValue('2');
  });
});
