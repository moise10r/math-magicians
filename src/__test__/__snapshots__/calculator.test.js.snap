// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page component should match the Calculator page 1`] = `
<div
  className="calculator-main-container"
>
  <div
    className="row flex-center result"
  >
    <input
      disabled={true}
      onChange={[Function]}
      type="text"
      value="0"
    />
  </div>
  <div>
    <div
      className="row flex-center "
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="row flex-center "
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="row flex-center "
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="row flex-center "
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row flex-center "
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
